### PR TITLE
Feature/don 14 captcha

### PR DESCRIPTION
**Pull request by Tokikko**
Added google captcha to signup page. Changed status code from 206 to 400 when a field is missing in submission form

### CODE REVIEW

File: utilities/googleReCaptcha.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+let got = require('got');
```
Comment (gnorbsl) [inline utilities/googleReCaptcha.js:1]: whats got used for here?

File: views/signup.ejs [hunk 2]
```diff
@@ -16,6 +16,13 @@
   <link rel="stylesheet" href="../css/signup.css">
   <!-- jquery -->
   <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.5.1/jquery.min.js"></script>
+  <!-- recaptcha -->
+  <script src="https://www.google.com/recaptcha/api.js" async defer></script>
+  <script>
+    function onSubmit(token) {
+      alert("hehe")
```
Comment (gnorbsl) [inline views/signup.ejs:23]: ?

File: utilities/googleReCaptcha.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+// check if captcha token is valid
+// returns boolean 
+async function validateReCaptchaToken(token) {
+    let googleUrl = `https://www.google.com/recaptcha/api/siteverify?secret=${process.env.GOOGLE_CAPTCHA_KEY}&response=${token}`;
+    return (async () => {
+        try {
+            const res = await superagent.post(googleUrl);
+            const responseBody = res.body;
+            if (responseBody.success === false) {
+                return false;
```
Comment (gnorbsl) [inline utilities/googleReCaptcha.js:13]: just return responseBody.success?

**Review by gnorbsl — CHANGES_REQUESTED** [review]
Please check if your master is up2date with the master here, it looks like you are missing quite a few things in your code

File: utilities/googleReCaptcha.js (hunk 3 above)
Comment (Tokikko) [reply utilities/googleReCaptcha.js:13]: Yeah i could, but i like to be more explicit

**Pull request closed without merging**